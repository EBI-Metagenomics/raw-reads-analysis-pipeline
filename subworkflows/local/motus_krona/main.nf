include { KRONA_KTIMPORTTEXT } from '../../../modules/ebi-metagenomics/krona/ktimporttext/main'
include { MOTUS_PROFILE } from '../../../modules/nf-core/motus/profile/main'
include { MOTUS2KRONA } from '../../../modules/local/motus2krona/main'

workflow MOTUS_KRONA {

    take:
    samples   // channel: [ val(meta), [ fastx ] ]
    motus_db  // path

    main:
    ch_versions = Channel.empty()

    MOTUS_PROFILE(
        samples,
        motus_db
    )
    ch_versions = ch_versions.mix(MOTUS_PROFILE.out.versions)
    // MOTUS_PROFILE.out.out.view { "motus.out - ${it}" }

    MOTUS2KRONA(MOTUS_PROFILE.out.out, file("${projectDir}/bin/motus2krona.py"))
    ch_versions = ch_versions.mix(MOTUS2KRONA.out.versions)

    // KRONA
    KRONA_KTIMPORTTEXT(MOTUS2KRONA.out.krona)
    ch_versions = ch_versions.mix(KRONA_KTIMPORTTEXT.out.versions)

    emit:
    versions = ch_versions                     // channel: [ versions.yml ]
}

