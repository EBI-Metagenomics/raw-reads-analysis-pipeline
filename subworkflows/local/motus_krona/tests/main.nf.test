nextflow_workflow {

    name "Test Subworkflow MOTUS_KRONA"
    script "../main.nf"
    workflow "MOTUS_KRONA"

    tag "subworkflows"
    tag "subworkflows_"
    tag "subworkflows/motus_krona"


    test("mini sample single end should match snapshot") {

        when {
            workflow {
                """
                input[0] = Channel.of(
		    [
		        ['id': 'test1', 'single_end': true, 'platform': 'wgs'],
	                file("${projectDir}/tests/fixtures/motus_krona/ERR10889056.fq", checkIfExists: true),
		    ]
		)
		input[1] = Channel.value(
		    file("${projectDir}/tests/fixtures/dbs/motus_mini/db_mOTU", checkIfExists: true)
		)
                """
            }
        }

        then {
	    def krona_out = workflow.out.krona.collect{ meta, fp -> [meta, file(fp).text.readLines().sort()] }

            assert workflow.success
            assert snapshot(
                krona_out,
		workflow.out.versions
	    ).match()
        }
    }

    test("mini sample paired end should match snapshot") {

        when {
            workflow {
                """
                input[0] = Channel.of(
		    [
		        ['id': 'test1', 'single_end': false, 'platform': 'wgs'],
	                [
			    file("${projectDir}/tests/fixtures/motus_krona/ERR10889056.fq", checkIfExists: true),
			    file("${projectDir}/tests/fixtures/motus_krona/ERR10889056_r.fq", checkIfExists: true),
			]
		    ]
		)
		input[1] = Channel.value(
		    file("${projectDir}/tests/fixtures/dbs/motus_mini/db_mOTU", checkIfExists: true)
		)
                """
            }
        }

        then {
	    def krona_out = workflow.out.krona.collect{ meta, fp -> [meta, file(fp).text.readLines().sort()] }

            assert workflow.success
            assert snapshot(
                krona_out,
		workflow.out.versions
	    ).match()
        }
    }

    test("mini sample with no matches should match snapshot") {

        when {
            workflow {
                """
                input[0] = Channel.of(
		    [
		        ['id': 'test1', 'single_end': true, 'platform': 'wgs'],
		        file("${projectDir}/tests/fixtures/motus_krona/ERR10889056_nomatch.fq", checkIfExists: true),
		    ]
		)
		input[1] = Channel.value(
		    file("${projectDir}/tests/fixtures/dbs/motus_mini/db_mOTU", checkIfExists: true)
		)
                """
            }
        }

        then {
	    def krona_out = workflow.out.krona.collect{ meta, fp -> [meta, file(fp).text.readLines().size()] }

            assert workflow.success
            assert snapshot(
                krona_out,
		workflow.out.versions
	    ).match()
        }
    }
}
