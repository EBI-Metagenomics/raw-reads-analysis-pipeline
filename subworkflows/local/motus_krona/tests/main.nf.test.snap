{
    "mini sample with no matches should match snapshot": {
        "content": [
            [
                [
                    {
                        "id": "test1",
                        "single_end": true,
                        "platform": "wgs"
                    },
                    0
                ]
            ],
            [
                "versions.yml:md5,19340de5d0deb51f0cc3f587d46ec41f",
                "versions.yml:md5,949ffb5623556e06c140de87118b5826",
                "versions.yml:md5,952c72664172faa8c853678f69e005c3"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-05-29T13:32:18.369086315"
    },
    "mini sample paired end should match snapshot": {
        "content": [
            [
                [
                    {
                        "id": "test1",
                        "single_end": false,
                        "platform": "wgs"
                    },
                    [
                        "1.0\tk__Bacteria\tp__Actinobacteria\tc__Actinobacteria\to__Bifidobacteriales\tf__Bifidobacteriaceae\tg__Bifidobacterium\ts__Bifidobacterium longum [Bifidobacterium longum CAG:69/Bifidobacterium longum]",
                        "1.0\tk__Bacteria\tp__Bacteroidetes\tc__Bacteroidia\to__Bacteroidales\tf__Bacteroidaceae\tg__Bacteroides\ts__Bacteroides thetaiotaomicron",
                        "1.0\tk__Bacteria\tp__Firmicutes\tc__Bacilli\to__Lactobacillales\tf__Lactobacillaceae\tg__Lactobacillus\ts__Lactobacillus gasseri"
                    ]
                ]
            ],
            [
                "versions.yml:md5,19340de5d0deb51f0cc3f587d46ec41f",
                "versions.yml:md5,949ffb5623556e06c140de87118b5826",
                "versions.yml:md5,952c72664172faa8c853678f69e005c3"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-05-29T13:27:00.810048403"
    },
    "mini sample single end should match snapshot": {
        "content": [
            [
                [
                    {
                        "id": "test1",
                        "single_end": true,
                        "platform": "wgs"
                    },
                    [
                        "2.0\tk__Bacteria\tp__Bacteroidetes\tc__Bacteroidia\to__Bacteroidales\tf__Bacteroidaceae\tg__Bacteroides\ts__Bacteroides thetaiotaomicron",
                        "2.0\tk__Bacteria\tp__Firmicutes\tc__Bacilli\to__Lactobacillales\tf__Lactobacillaceae\tg__Lactobacillus\ts__Lactobacillus gasseri",
                        "3.0\tk__Bacteria\tp__Actinobacteria\tc__Actinobacteria\to__Bifidobacteriales\tf__Bifidobacteriaceae\tg__Bifidobacterium\ts__Bifidobacterium longum [Bifidobacterium longum CAG:69/Bifidobacterium longum]"
                    ]
                ]
            ],
            [
                "versions.yml:md5,19340de5d0deb51f0cc3f587d46ec41f",
                "versions.yml:md5,949ffb5623556e06c140de87118b5826",
                "versions.yml:md5,952c72664172faa8c853678f69e005c3"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-05-29T13:22:03.524822248"
    }
}